### PR TITLE
DOC: remove hidden stats sections from sidebar/toctree

### LINKED ISSUE #17866: DOC: Should `Result Classes` be its own top level section?
### Issue with current documentation:

Currently the SciPy documentation for version 1.10 shows [Result Classes](https://docs.scipy.org/doc/scipy/reference/stats._result_classes.html) as its own, top level, section. I don't believe this is correct, as they are only results for stats. Should this come under scipy.stats.result_classes?

### Idea or request for content:

Move results classes to scipy.stats.result_classes

### Additional context (e.g. screenshots, GIFs)

 SciPy Result Classes in wrong position

### PR DESCRIPTION
**Pull request by tupui**
Closes #17866

Remove the result classes and contingency tables from the toctree.

(I also always have a linalg error in an example locally, so I fixed it)

### CODE REVIEW

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:575]: One might wonder why they are documented at all then; perhaps we should explain. Also, rather than issuing a command to the user, consider phrasing it in terms of what SciPy will or will not support.

> These classes are private, but they are included here because instances of them are returned by other statistical functions. User import and instantiation is not supported.

File: scipy/signal/_signaltools.py
Comment (mdhaber) [inline scipy/signal/_signaltools.py:4111]: What was the warning? Anything to worry about?

**Review by mdhaber — COMMENTED** [review]
I'll take a look at how this renders. Just a phrasing suggestion, but I can merge as-if if you prefer. 

@Kai-Striega were you also working on this?

File: scipy/signal/_signaltools.py
Comment (tupui) [reply scipy/signal/_signaltools.py:4111]: I think it's ok it was: `numpy.linalg.LinAlgError: SVD did not converge in Linear Least Squares`

File: scipy/stats/__init__.py
Comment (tupui) [reply scipy/stats/__init__.py:575]: Let's do that 👍
```suggestion
.. warning::

   These classes are private, but they are included here because instances
   of them are returned by other statistical functions. User import and
   instantiation is not supported.
```

**Review by tupui — COMMENTED** [review]
Thanks Matt. Yes let's wait for Kai in case he had a different or complementary approach. (I just spend a few hours fighting with Sphinx so wanted to push what I had 😅)

**Comment by tupui** [comment]
@mdhaber I am tempted to clean the sidebar more. Another PR I guess?

**Comment by mdhaber** [comment]
I agree. It's fine to do that here since we need to commit wording suggestion anyway..

 But let's just remove the `stats` ones. We can encourage primary maintainers of other subpackages to do the same, but in some cases it will make sense to keep them (e.g. the submodules of `scipy.cluster` seem to be _all_ that is in `scipy.cluster`, so it might make sense to_only_ include those in the sidebar).

**Comment by tupui** [comment]
Done, I will send an email to the mailing list for the other modules.

File: scipy/signal/_signaltools.py
Comment (Kai-Striega) [reply scipy/signal/_signaltools.py:4111]: I don't think we need to worry about this. It's just that for *some* random seeds, the calculation is less stable. See: https://github.com/numpy/numpy/issues/22664

**Review by Kai-Striega — APPROVED** [review]
LGTM; message looks good and I don't think avoiding the warning is going to cause any issues.

I was trying to find a similar solution but I think you both beat me to it and actually got it done so kudos to you

**Comment by tupui** [comment]
Thanks @Kai-Striega! If you want I can let you do the toctree change I proposed on the mailing list. LMK

**Comment by Kai-Striega** [comment]
That's ok, @tupui, but I don't think I'll have the time to battle with Sphinx. I've bitten off quite a few issues this weekend so I think I'll focus on those

**Comment by tupui** [comment]
Ok sure I will do this on Monday then 👍 (FYI please don't merge anything until the linter PRs are in)

**Comment by mdhaber** [comment]
Oops hope that didn't cause problems. I thought you already merged the "lint everything" PR, but I guess it was just the pre-commit hook. Well, fortunately the "lint everything" PR still has no conflicts with main.

**Comment by tupui** [comment]
haha yep should be fine

**Pull request merged**
